### DISCUSSION TITLE
Trino completion for functions?

### DISCUSSION THREAD

**Discussion by Samrose-Ahmed** [DTStack/monaco-sql-languages#77] [Q&A] [answered]
I see the completion suggests keywords for Trino, how can we get it to suggest built in functions like `date_trunc` etc, I see they are registered for highlighting.

**Comment by HaydenOrz** [comment, marked as answer]
The keyword list is derived from dt-sql-parser, and  keyword list of  SQL  will never change. 

However, for functions, user can use `create function` statement to create them, so we don't have built-in function name codeCompletion items for the time being. 

We're considering whether to add built-in functions to the default autocomplete items, but that hasn't been done yet.

Currently, you can only complete function names by customizing codeCompletion items.  Please see  [usage doc ](https://github.com/DTStack/monaco-sql-languages#usage) -> `Setup language features` -> `completionService `
